parsers/src/main/java/org/iqu/parsers/HtmlEventsParser.java

```diff
@@ -41,35 +41,46 @@
   private static String pattern = "yyyy-MM-dd'T'HH:mm:ss'+'hh:mm";
   private static DateFormat dF = new SimpleDateFormat(pattern);
   private String categories = "";
-  private String source;
+  private String sourceURL;
+  private Source source;
 
   @Override
   public Source getSource() {
-    // TODO: implement reading of source information from feed
-    return null;
+    return source;
   }
 
   @Override
   public List<Event> readFeed(String sourceURL, String encoding) {
 
     List<Event> events = new ArrayList<Event>();
-    source = sourceURL;
+    this.sourceURL = sourceURL;
     Document doc = null;
     try {
       doc = Jsoup.connect(sourceURL).get();
```

Comment:
change to, because otherwise you will get a NPE :   

con =  Jsoup.connect(sourceURL);
if(doc!= null ) { 
       doc = con.get();
       readSource(doc);
       readItems(events, doc);
}